src/main/java/com/app/eventOrganizer/service/UserDataService.java

```diff
@@ -0,0 +1,26 @@
+package com.app.eventOrganizer.service;
+
+import com.app.eventOrganizer.model.UserModel;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import com.app.eventOrganizer.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserDataService {
+    private final UserRepository userRepository;
+
+    public UserModel getAuthenticatedUsersID() {
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String userEmail = authentication.getName();
+        Optional<UserModel> userModelOptional = userRepository.findByEmail(userEmail);
+        UserModel userModel = userModelOptional.get();
+
+        return userModel;
```

Comment:
These 5 lines can be expressed in 2:

```suggestion
        String userEmail = SecurityContextHolder.getContext().getAuthentication().getName();
        return userRepository.findByEmail(userEmail).orElseThrow();
```

Reply:
get it, done